### PR TITLE
Automatic changelog generation

### PR DESCRIPTION
**Pull request by danybony**
When a new tag is found (`check_for_new_build_label.sh`), after emails have been sent, the changelog generation starts (`generate_last_changelog.sh`):
1. repo init to the last tag
2. repo sync
3. changelog generation in the AOSP working dir

This PR doesn't cover the changelog upload

### CODE REVIEW

**Comment by danybony** [comment]
:eyes: @charroch @blundell

**Comment by blundell** [comment]
how come these weren't committed earlier? :+1:

File: README.md [hunk 1]
```diff
@@ -3,6 +3,16 @@ Receive a notification email every time a new git tag is found in AOSP<br/>
 _Will_ generates a change log between different aosp tags. 
 
 ### Usage
+#### New tag notification
 ```
-$ ./check_for_new_build_label.sh <notification email address>
+$ ./check_for_new_build_label.sh <notification email address> <AOSP working directory>
+```
+
+The param `AOSP working directory` is optional and must specify the *absolute* path of the directory in which will be cloned the whole AOSP code.
```
Comment (rock3r) [inline README.md]: `must specify the *absolute* path of the directory in which the whole AOSP code will be cloned.` sounds better to me

File: README.md [hunk 2]
```diff
@@ -3,6 +3,16 @@ Receive a notification email every time a new git tag is found in AOSP<br/>
...
+If this param is specified, when a new tag is found then the related changelog html page is automatically generated
```
Comment (rock3r) [inline README.md]: `If this parameter is passed, when a new tag is found on the remote the creation of the corresponding changelog HTML page is started automatically.`

**Comment by danybony** [comment]
@blundell lack of spare time unfortunately.. :(

File: README.md (hunk 2 above)
Comment (danybony) [inline README.md]: :+1:

File: README.md (hunk 1 above)
Comment (danybony) [inline README.md]: Yeah, your sounds more English :tongue:

File: generate_last_changelog.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+AOSP_DIRECTORY=$1
+WORK_DIRECTORY=./aosp/checkouts/build
+
+# Prepare the AOSP working directory with the templates and executables used for the changelof generation
```
Comment (rock3r) [inline generate_last_changelog.sh]: Typo: _used for the **changelof** generation_

File: generate_last_changelog.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+TAGS_ARRAY=(`echo $LAST_TAGS | tr "," "\n"`)
+TARGET_TAG=${TAGS_ARRAY[0]}
+PREVIOUS_TAG=${TAGS_ARRAY[1]}
+
+# Update the AOSP working directory with a repo sync to the target tag
+cd $AOSP_DIRECTORY
+repo init -u https://android.googlesource.com/platform/manifest -b $TARGET_TAG
+repo sync
+
+./get_gitlog.sh $PREVIOUS_TAG $TARGET_TAG
```
Comment (rock3r) [inline generate_last_changelog.sh]: Missing empty line at EOF :tongue:

File: generate_last_changelog.sh (hunk 3 above)
Comment (danybony) [inline generate_last_changelog.sh]: :+1:

File: generate_last_changelog.sh (hunk 4 above)
Comment (danybony) [inline generate_last_changelog.sh]: :+1:

File: html_templates/footer.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+		</div>
+	</div>
+</div>
+</div>
+
+<div id="footer">
+  <div class="container">
+  	<div class="row">
+	<div class="col-md-6">
+    <p class="text-muted">&copy; Copyright 2015 Novoda Ltd., All Rights Reserved.</p>
```
Comment (rock3r) [inline html_templates/footer.html:11]: Why not populating the year dynamically like you do for the header?

Comment (charroch) [inline html_templates/footer.html:11]: This is copyright so it should not really.

Comment (rock3r) [inline html_templates/footer.html:11]: What I mean is, will you remember changing the footer in 8 months with the new year? Why not having this filled in with the year at time of generating the diff?

**Pull request merged**